account_invoice_mass_sending/models/account_move.py

```diff
@@ -14,56 +13,51 @@ class AccountInvoice(models.Model):
         "and it will prevent the sending of a duplicated mail.",
     )
 
-    def mass_send_print(self):
+    def mass_sending(self, template=None):
         """
-        This method triggers the asynchronous "Send & Print" for the selected
-        invoices for which there is no asynchronous sending in progress.
+        This method triggers the asynchronous sending for the selected
+        invoices for which there is no asynchronous sending in progress
+        and an email address is defined.
         """
-        invoices_to_send = self.filtered(lambda i: not i.sending_in_progress)
-        in_progress_invoices_count = len(self) - len(invoices_to_send)
-        title = _("Invoices: Send & Print")
-        if in_progress_invoices_count > 0:
-            warn_msg = _(
-                "The sending of {in_progress_invoices_count} invoices is "
-                "already in progress."
-            ).format(in_progress_invoices_count=in_progress_invoices_count)
-            self.env.user.notify_info(title=title, message=warn_msg)
+        invoices_to_send = self.filtered(
+            lambda i: not i.sending_in_progress and i.partner_id.email
+        )
         if invoices_to_send:
-            msg = _(
-                "The sending of {invoices_count} invoices will be processed "
-                "in background."
-            ).format(invoices_count=len(invoices_to_send))
-            self.env.user.notify_info(title=title, message=msg)
-            invoices_to_send.write({"sending_in_progress": True})
-            invoices_to_send.with_delay().do_prepare_send_print()
-
-    def do_prepare_send_print(self):
-        for rec in self:
-            rec.with_delay().do_send_print()
-
-    def do_send_print(self):
-        for rec in self:
-            if not rec.partner_id.email:
-                raise UserError(
-                    _("Missing email address on customer {customer_name}.").format(
-                        customer_name=rec.partner_id.display_name
-                    )
-                )
-            action_invoice_wizard = rec.action_invoice_sent()
-            ctx = action_invoice_wizard["context"]
-            ctx.update(
+            invoices_to_send.write(
                 {
-                    "active_id": rec.id,
-                    "active_ids": rec.ids,
-                    "active_model": "account.invoice",
+                    "sending_in_progress": True,
                 }
             )
-            invoice_wizard = (
-                self.env[action_invoice_wizard["res_model"]]
-                .with_context(ctx)
-                .create({})
-            )
-            invoice_wizard._compute_composition_mode()
-            invoice_wizard.onchange_template_id()
-            invoice_wizard.send_and_print_action()
-        self.write({"sending_in_progress": False})
+            for invoice in invoices_to_send:
+                description = _("Send invoice {} by email").format(invoice.name)
```

Comment:
I normally use this kind of translation https://www.odoo.com/documentation/14.0/developer/howtos/translations.html#variables, since you take benefit of this fragment of code
![image](https://github.com/OCA/account-invoicing/assets/76053579/60afa5c0-6ff0-45ea-b515-fea483c29ac8)

Reply:
Thanks for the info, not sure I understand what you mean though. You could suggest change.

Reply:
I get it :)
Also did some research and the format() could be dangerous
https://github.com/OCA/pylint-odoo/issues/302